### ISSUE TITLE
OutOfBoundsException on generating sequence.

### ISSUE THREAD

**Issue by MortalFlesh** [giorgiosironi/eris#117]
Hello,

I'm trying to test my `Tuple` class and I constantly get `OutOfBoundsException : Evaluation ratio XXX is under the threshold 0.5`.

But I also have test `testShouldCreateTupleFromAnyArray` ([see test](https://github.com/MortalFlesh/MFCollectionsPHP/commit/e8493549896787c8fa636ddfff58b014987eb209#diff-43cab33f29849dcd8cca65eff63056e5R18)) which works fine. So I'm confused what am I doing wrong.

Broken test are here:
_They are same but differs in the generation_
- [see broken](https://github.com/MortalFlesh/MFCollectionsPHP/commit/e8493549896787c8fa636ddfff58b014987eb209#diff-43cab33f29849dcd8cca65eff63056e5R45)
- [see broken 2](https://github.com/MortalFlesh/MFCollectionsPHP/commit/e8493549896787c8fa636ddfff58b014987eb209#diff-43cab33f29849dcd8cca65eff63056e5R102)

Basically I want to test, that my `Tuple` could be created from any array, transformed to string and then parsed again to originally given array. 
I would prefer the `broken 2` test, since it is more complex and also easier to read.

_Note: I don't want to test `floats` in the broken tests. As I do in the `working` test._

Any suggestions please?

Thank you <span class="emoji-outer emoji-sizer"><span class="emoji-inner" style="background: url(chrome-extension://immhpnclomdloikkpcefncmfgjbkojmh/emoji-data/sheet_apple_32.png);background-position:59.9882491186839% 77.96709753231492%;background-size:5418.75% 5418.75%" data-codepoints="1f60a"></span></span>

**Comment by MortalFlesh** [comment]
Well I finally end up with this:
```php
try {
    // .. test ..
} catch (\OutOfBoundsException $e) {
    $this->markTestSkipped($e->getMessage());
}
```
 ([see test](https://github.com/MortalFlesh/MFCollectionsPHP/pull/36/commits/90b7b18a7a33ce982769d139fab2591ffb99a6a6#diff-43cab33f29849dcd8cca65eff63056e5R70)).

And at least most of the time test passes. But I still got the error sometimes...

**Comment by giorgiosironi** [comment]
`Evaluation ratio` errors are due to Eris generating a large number of inputs, but most of them being discarded, usually because they do not satisfy `when()` clauses. Can you try using the [log](http://eris.readthedocs.io/en/latest/listeners.html#log) listener to understand what values are being generated and possibly discarded?

**Comment by ilario-pierbattista** [comment]
No response. Closing.
